### PR TITLE
SD-129: write integration tests for IPFS

### PR DESCRIPTION
**Pull request by domagojlatecki**
Added integration tests for IPFS that run against `ipfs-go` Docker container. This task should have also included integration tests for Web3/Geth, however for that I will need Solidity source code (or BIN files) in order to properly set-up local test blockchain with contracts. I have opened follow-up tasks for that in backlog.

### CODE REVIEW

File: src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                    assertThat(it.content).isEqualTo(fileContent.toByteArray())
+                }
+        }
+    }
+
+    @Test
+    fun `must return error for non-existent IPFS binary file in a directory`() {
+        verify("GatewayIpfsRepository implementation is used") {
+            assertThat(repository.javaClass)
+                .isEqualTo(GatewayIpfsRepository::class.java)
```
Comment (githinho) [inline src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt:194]: Would it make sense to put this assert in `@BeforeEach`? `GatewayIpfsRepositoryIntegTest` is called the test class so this assert in before each would look good to me.

File: src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/LocalIpfsRepositoryIntegTest.kt [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import com.ampnet.auditornode.testcontainers.IpfsTestContainer.uploadFileToIpfsDirectory
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import io.micronaut.test.support.TestPropertyProvider
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import javax.inject.Inject
+
+@MicronautTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class LocalIpfsRepositoryIntegTest : IntegTestBase(), TestPropertyProvider {
```
Comment (githinho) [inline src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/LocalIpfsRepositoryIntegTest.kt:26]: The same as for class `GatewayIpfsRepositoryIntegTest`, maybe before each, see what suits you better.

File: src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt (hunk 1 above)
Comment (domagojlatecki) [reply src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt:194]: What do you thing about also naming the annotated method `verify that GatewayIpfsRepository implementation is used`?

Comment (githinho) [reply src/integTest/kotlin/com/ampnet/auditornode/persistence/repository/GatewayIpfsRepositoryIntegTest.kt:194]: Even better!

**Review by githinho — APPROVED** [review]

**Pull request merged**